upkeep/369: Standardize PHPCS and PHPCompatibility GH Actions

faisal-alvi opened the pull request:
### All Submissions:

* [ ] Does your code follow the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

* Used the same GitHub action with the same PHPCS config (rules) which is used for all extensions when opening a PR
* Checking all PHPCS rules only changed lines in the PR
* Excluded PHPCompatibility ruleset from the phpcs config
* Check only the PHPCompatibility rule for all files in the PR
* Add/update and use `phpcs.xml.dist` and `phpcs-compat.xml.dist` config files.
* Validate PHPCS checks passing before merging a PR
* The ESLint workflow file is also updated to match w/ the PHPCS workflow file format.
* Updated npm run script for phpcompat

Closes #369.

### How to test the changes in this Pull Request:

All the GitHub Actions should be passed.

### Other information:

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Dev - Update PHPCS and PHPCompatibility GitHub Actions.

faisal-alvi committed:
update the phpcs action file

faisal-alvi committed:
add phpcompat file

faisal-alvi committed:
add phpcs.sh file

faisal-alvi committed:
update php to 7.4

faisal-alvi committed:
update the php in script

faisal-alvi committed:
update the php compat action

faisal-alvi committed:
"GPL-2.0+" is depcrecated, used "GPL-2.0-or-later"

faisal-alvi committed:
remove the phpcs.sh file

faisal-alvi committed:
add an xml dist file

faisal-alvi committed:
update the action file to use the config

faisal-alvi committed:
add a file doc

faisal-alvi committed:
test with the static command

faisal-alvi committed:
add a package name

faisal-alvi committed:
print the tested file as dots

faisal-alvi committed:
if: changed-files.outputs.any_changed is true

faisal-alvi committed:
fix minor phpcs issues

faisal-alvi committed:
in_array 3rd param added

faisal-alvi committed:
trying to check any_changed value

faisal-alvi committed:
trying to get the changed file in the command

faisal-alvi committed:
remove if condition

faisal-alvi committed:
adding back the conditions to run only when found

faisal-alvi committed:
phpcs fixes

faisal-alvi committed:
update the eslint to match w/action files format

faisal-alvi committed:
update the eslint file to run for js file changes

faisal-alvi committed:
changed phpcs.xml to phpcs.xml.dist

faisal-alvi committed:
composer update

faisal-alvi committed:
changed phpcs.xml to phpcs.xml.dist

faisal-alvi committed:
remove phpcs.xml export-ignore

faisal-alvi committed:
update add phpcs-compat.xml.dist file

faisal-alvi committed:
update scripts with updated commands

dkotter reviewed (commented):
Overall this looks good but I don't think we should be bumping our minimum PHP version here. I believe the goal is to still ensure our minimum PHP version matches the minimum PHP version of the minimum Woo version we support (which is still 7.3). The know the linked issue and the P2 post mention 7.4 but I'd suggest we stick with 7.3 for now

faisal-alvi committed:
reverting to PHP 7.3 from 7.4

dkotter commented on .github/workflows/qit.yml:
```diff
@@ -54,7 +54,7 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
-          php-version: 7.4
+          php-version: 7.3
```
This needs to stay at 7.4, as that's the minimum required for QIT
```suggestion
          php-version: 7.4
```

dkotter committed:
Update PHP version

faisal-alvi committed:
exclude entire PHPCompatibilityWP ruleset

faisal-alvi committed:
run PHPCS only on changed lines, not changed files

faisal-alvi committed:
fix phpcs yml broke issue

faisal-alvi committed:
add sirbrillig/phpcs-changed as dev dep

faisal-alvi committed:
composer update

faisal-alvi committed:
parent d9fd52a569a7bd74caed485ea8836b6a21cc5743
author faisal-alvi <faisalalvi33@gmail.com> 1695117777 +0530
committer faisal-alvi <faisalalvi33@gmail.com> 1695128028 +0530

again: intensional phpcs error to see changed line action

fix space issue

fix fatal: Not a valid object name origin/c5534d8eaaf00fa73e868aa4f126650060cea7e6

getting github variables to get the branch name

echo in existing run command

try to see variable values

fix yml broke

generating error to see line changes in phpcs

fixed phpcs strict comp issue

faisal-alvi committed:
update the correct path of phpcs-changed

faisal-alvi committed:
fix spacing issues

faisal-alvi committed:
update the comment msg

faisal-alvi committed:
remove the comment msg

dkotter commented on .gitattributes:
```diff
@@ -14,7 +14,7 @@ phpunit.xml.dist export-ignore
 webpack.config.js export-ignore
 postcss.config.js export-ignore
-phpcs.xml export-ignore
+phpcs.xml.dist export-ignore
```
We probably want to exclude the PHPCS compat config as well I think

dkotter commented on composer.json:
```diff
@@ -13,6 +13,7 @@
     "require-dev": {
         "phpunit/phpunit": "^8.0",
         "woocommerce/woocommerce-sniffs": "^0.1.3",
+		"sirbrillig/phpcs-changed": "^2.11.1",
```
Looks like spacing is a bit off here

dkotter commented on phpcs-compat.xml.dist:
```diff
@@ -0,0 +1,16 @@
+    <rule ref="PHPCompatibility"/>
+
+    <!-- Files to check -->
+	<arg name="extensions" value="php"/>
```
Looks like spacing is a bit off here

faisal-alvi committed:
exclude the PHPCS compat config

faisal-alvi committed:
fixed spacing in composer file

faisal-alvi committed:
spacing fixed in compat config

jeffpaul commented on .github/workflows/phpcs.yml:
```diff
@@ -1,30 +1,50 @@
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v35
```
Shouldn't this be changed lines and not changed files?

dkotter replied:
It uses the `changed-files` action to get a list of changed files and then passes those files into the `phpcs-changed` package (line 50 in this file) which then pulls the changed lines out of those files

faisal-alvi committed:
bump `woocommerce-sniffs` to `1.0.0`

faisal-alvi commented:
Just added a new commit to bump `woocommerce-sniffs` to `1.0.0` ([internal slack discussion link](https://10up.slack.com/archives/C01UCQGPBDJ/p1696260699336079?thread_ts=1696009651.377169&cid=C01UCQGPBDJ))
